### PR TITLE
feat(game): guilds

### PR DESCRIPTION
**Pull request by MeikelLP**
Supersedes #12 

- [x] Create Guilds
- [x] Delete Guilds
- [x] Invite To Guilds
- [x] Leave Guilds
- [x] Guild Leaders
- [x] Guild Roles
- [x] View Guild Info
  - [x] Guild News
  - [x] Guild Members
  - [x] Guild Ranks
- [x] Friend List Guild Members
- [x] Permissions
- [x] Spend Exp / Guild Leveling

## Additional changes

* feat: Allow constants and calculations in `FieldAttribute.Length`
* fix: not able to inject scoped service into command handlers

### CODE REVIEW

File: src/Executables/Game/GuildManager.cs [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            .Where(x => x.Id == leaderId)
+            .ExecuteUpdateAsync(setter => setter
+                .SetProperty(p => p.GuildId, guild.Id), token);
+
+        return new GuildData
+        {
+            Id = guild.Id,
+            Name = name,
+            Level = 1,
+            MaxMemberCount = 10,
```
Comment (uutkukorkmaz) [inline src/Executables/Game/GuildManager.cs:92]: Magic number 👀

also, why don't we get the `GuildData.MaxMemberCount` from `guild.MaxMemberCount`?

**Review by uutkukorkmaz — COMMENTED** [review]
Oh boy, that was a long read :) 
I also tested and it looks good. 👍

**Comment by MeikelLP** [comment]
> Oh boy, that was a long read :) I also tested and it looks good. 👍

It's fine if you don't read all of it. It's a lot but I thought that's better than something half baked. Also at some points I didn't care too much about performance or code quality but rather getting the feature ready. Polishing can be done in subsequent PRs/features/fixes 👍

File: src/CorePluginAPI/Game/Guild/GuildData.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Immutable;
+
+namespace QuantumCore.API.Game.Guild;
+
+public class GuildData
+{
+    public uint Id { get; set; }
+    public string Name { get; set; } = "";
```
[the remark this answers is not in the document]
Comment (MeikelLP) [reply src/CorePluginAPI/Game/Guild/GuildData.cs:8]: I tried to look into the guild item but networkwise nothing is send from the client when selecting an icon. I'd put it in an additional feature issue to implement later 👍

File: src/Executables/Game/GuildManager.cs (hunk 1 above)
Comment (MeikelLP) [reply src/Executables/Game/GuildManager.cs:92]: That's a bug. Well spotted 👍

**Review by uutkukorkmaz — APPROVED** [review]

**Pull request merged**